setupServers.js

```diff
@@ -0,0 +1,157 @@
+import * as fs from 'fs';
+import restify from 'restify';
+import {setupRoutes} from './routes';
+const {dialog} = require('electron');
+import sudo from 'electron-sudo';
+import {replace} from 'ramda';
+
+const httpsMessage = 'This application will establish an encrypted link ' +
+    'between the connector application and plotly 2.0 client. In order to ' +
+    'provide that communication tunnel, a new private key for your device ' +
+    'will be generated using the \'openssl\' command which requires ' +
+    'administrator\'s password.';
+
+// asking for sudo more than once, but show a dialog window only once
+let messageShown = false;
+const showSudoMessage = () => {
+    if (!messageShown) {
+        messageShown = true;
```

Comment:
having these variables as part of the module scope is a bit of an anti-pattern because importing anything from this module has side-affects: if you import it twice, you'll reset the `messageShown` variable.

Instead, make it an object and instantiate it where you need it.

```
SudoMessageHandler = () => {
    let messageShown = false;
    return {
        showMessage: () => {
              if(!messageShown) {
                    messageShown = true;
                    dialog.showMessageBox(...)
              }
    }
}

[...]

// then later in your code
sudoMessageHandler = SudoMessageHandler();
sudoMessageHandler.showMessage(...)
```